### ISSUE TITLE
Drag and drop doesn't work anymore with Epiphany/GNOME Web

### ISSUE THREAD

**Issue by nekohayo** [fguillot/kanboard#1382]
For some reason, sometime in recent months the ability to drag a card (from on column/lane/position to another) stopped working with https://en.wikipedia.org/wiki/Web_(web_browser). Instead of moving the card, it tries selecting the text.

Used to work fine, and still works with Firefox.

**Comment by sparkles645** [comment]
I can relate, ref #1380

**Comment by nekohayo** [comment]
Not 100% identical; in my case I can't even unstick/get the card "off the ground", it just selects the text.

**Comment by sparkles645** [comment]
I see.
Something has changed, but I can't really pinpoint what and when.
In what **version of KB did you first experience it?**

**Comment by nekohayo** [comment]
Thankfully I have two instances so we can "roughly" bisect (without really bisecting) :

* One instance was not upgraded in a long time (2015-05-02) and does not exhibit the behavior; it is sitting at commit fa59a1487c01b
* The other is one version behind (2015-08-30), sitting at commit de91d5820b9f and it is affected by this bug, as you can see in this screencast: http://jeff.ecchi.ca/public/kanboard-1382.webm

Interestingly, while recording that screencast I hit a very rare corner case where there's a very tiny hitbox where the drag & drop works, but I don't know where it is exactly.

**Comment by sparkles645** [comment]
Referencing your dates, (2015-05-02) means  >= v 1.0.014(?) - and (2015-08-30) means >=1.0.18 - according to <http://kanboard.net/news>

**Comment by fguillot** [comment]
@nekohayo Do you have any Javascript errors in the browser console?

I don't really test Kanboard with Epiphany but if I remember correctly it's based on webkit so it should work.

**Comment by nekohayo** [comment]
Yeah, it should work… in theory ;) this is Epiphany 3.16 with WebKit 2.8.5. Again, it works with the older kanboard release, just not the new one. No errors in the inspector's console.

**Comment by sparkles645** [comment]
Removing my comments as my issue is different and reason for it is discovered

**Comment by aquimi** [comment]
The cards are blocked to move more than 5 times to any lane.
Please tell me if there is already a solution to this problem, we should be grateful.

**Comment by aquimi** [comment]
please help me

**Comment by aquimi** [comment]
???

**Comment by nekohayo** [comment]
@aquimi this is not the right place, this bug report about Epiphany has nothing to do with your issue. Open a separate ticket if you need to.

**Comment by nekohayo** [comment]
Just for the sake of completeness wrt the original issue: I have now tested Epiphany 3.20 with WebkitGTK 2.10.x, and the issue persists. Technically, it's not strictly the same as Blink/Webkit, so if you have time @fguillot it would be worth installing that browser to give it a quick shot and confirm the presence of the bug.

**Comment by nekohayo** [comment]
I _think_ I may have found the "issue"... in Firefox the whole card is a draggable element, but in Epiphany the cards can only be dragged with the little pictogram/icon in their top-left corner, the one with four arrows that look like a visual metaphor for "fullscreen". This icon does not appear in Firefox.

My question is thus:
* Why the difference in behavior across browsers?
* Why would the "only the icon is draggable" behavior be desirable vs the traditional approach where the whole card is a drag source?

**Comment by sparkles645** [comment]
ref #1142 for background
* Icon meant as "Move icon"

**Comment by nekohayo** [comment]
Ok then this sounds like user agent misdetection... Epiphany/WebKit will claim to be

```Mozilla/5.0 (Fedora; Linux x86_64) AppleWebKit/602.1 (KHTML, like Gecko) Version/8.0 Safari/602.1 Epiphany/3.18.0```

Which is bullshit as we know, but apparently there's "no choice":
* https://bugzilla.gnome.org/show_bug.cgi?id=721191
 * https://bugs.webkit.org/show_bug.cgi?id=124229
* https://bugzilla.gnome.org/show_bug.cgi?id=753610
 * https://bugs.webkit.org/show_bug.cgi?id=142074

Probably a dozen other User Agent issues exist... though I'm sure your input could be helpful for those bug reports.

**Comment by sparkles645** [comment]
* I reverted back to 1.0.16 for the (personal) instance that I use with a tablet.
* The tablet browser (Firefox, Android) triggered the "Move Icon" - and I tried, but didn't get used to it.

**Comment by fguillot** [comment]
- The draggable icon you are talking about should be visible only on touch devices. It's a quick and dirty workaround to avoid accidental moves on tablets, otherwise you move tasks if you need to scroll the viewport. I'm aware that it's not the best solution, but that works with Jquery sortable...
- Now, the detection of touch devices is not very smart because that use only the jQuery helper `$.support.touch`, but that doesn't take into consideration if your machine is compatible with touch and pointer events. So you may see the icon even on a laptop (I cannot test with all possible devices of the world).

Can you confirm those assumptions:
- Running this command `$.support.touch` in Epiphany browser console returns true
- Are you using a laptop compatible with touch and mouse events?

**Comment by nekohayo** [comment]
> Running this command $.support.touch in Epiphany browser console returns true

In my case it returns "undefined". At the bottom of the webkit inspector in Epiphany there seemed to be a console so I tried entering that command, but noticed that while $ and support were autocompleted, touch was not. So there is no "touch" among the choices in .support...

My Thinkpad X201t is indeed touch and stylus-enabled, however the issue also occurs on my desktop computer (which has no notion of touch whatsoever).

**Comment by fguillot** [comment]
Ok, I will install a vm with Fedora/Gnome when I will have 5min to investigate.

**Comment by nekohayo** [comment]
I thought it would be easier to just install epiphany (or "epiphany-browser") under your distro of choice (especially as VM's aren't the fastest thing ever and you would still have to install the epiphany package after installing the OS), unless you're stuck on a LTS or something, but that's up to you ;)

**Comment by fguillot** [comment]
I tested quickly with Epiphany on the last Fedora and the browser indicate that it supports touch events even if I'am on device that doesn't support touch events. I don't know why they have done that, a lot of stuff on the web will probably break...

Now, the next step is to find a workaround for this issue...

**Comment by nekohayo** [comment]
You say it indicates that it supports touch events... how is that so, given that on my end ```$.support.touch``` always returned "undefined"?

**Comment by fguillot** [comment]
It's returning true for me, event `'ontouchstart' in window` or other methods are returning true.

**Comment by nekohayo** [comment]
Here is the issue in WebKitGtk, feel free to join the discussion: https://bugs.webkit.org/show_bug.cgi?id=139681

**Comment by lathan** [comment]
I was facing the same issue on only one particular instance of Firefox. When I used a new Firefox profile I didn't face the issue. So it was obviously some setting being the culprit. The jquery `$.support.touch` mentioned above returned `true` on the problematic Firefox profile and `false` on the clean profiles. After searching a bit I went into `about:config` and found that there was a user-set `dom.w3c_touch_events.enabled;1` which I promptly reset to the default of `dom.w3c_touch_events.enabled;0`. This resolved the issue. I know it's not directly related to the Epiphany issue, but I thought it was worth mentioning.

Cheers
